src/border_agent/border_agent.cpp

```diff
@@ -504,7 +543,7 @@ std::string BorderAgent::GetAlternativeServiceInstanceName() const
     uint16_t                                rand = uniform_dist(engine);
     std::stringstream                       ss;
 
-    ss << BaseServiceInstanceName() << " (" << rand << ")";
+    ss << mServiceInstanceName << " (" << rand << ")";
```

Comment:
Note that `GetAlternativeServiceInstanceName()` could be called multiple times when generating the alternative name. This could result in appending the random suffix multiple times to the instance name.
```
// on conflict:
mServiceInstanceName = GetAlternativeServiceInstanceName();
```